src/components/create-appointment-fourth/CreateAppointmentFourthAutocomplete/CreateAppointmentFourthAutocomplete.tsx

```diff
@@ -0,0 +1,156 @@
+import Location from 'src/assets/icons/Location';
+import { useLocales } from 'src/locales';
+import usePlacesAutocomplete from 'use-places-autocomplete';
+import { AutocompletedLocation } from '../types';
+import {
+  PLACES_API_KEY,
+  PLACES_COUNTRY_RESTRICTIONS,
+  PLACES_DEBOUNCE_DELAY,
+  PLACES_SCRIPT_LIBRARIES,
+} from './constants';
+import { getPlaceIdDetails } from './helpers';
+import { ErrorMessage, StyledGrid } from './styles';
+
+interface CreateAppointmentFourthAutocompleteProps {
+  onLocationChange: (newLocation: AutocompletedLocation) => void;
+}
+
+export default function CreateAppointmentFourthAutocomplete({
+  onLocationChange,
+}: CreateAppointmentFourthAutocompleteProps): JSX.Element | null {
+  const {
+    currentLang: { value: placesAutocompleteLanguage },
+  } = useLocales();
+  const { isLoaded } = useLoadScript({
+    googleMapsApiKey: PLACES_API_KEY,
+    libraries: PLACES_SCRIPT_LIBRARIES,
+    language: placesAutocompleteLanguage,
+  });
+
+  if (!isLoaded) return null;
+
+  return <PlacesAutocomplete onLocationChange={onLocationChange} />;
+}
+
+function PlacesAutocomplete({
+  onLocationChange,
+}: CreateAppointmentFourthAutocompleteProps): JSX.Element {
+  const { translate } = useLocales();
+  const {
+    value: location,
+    setValue: setLocation,
+    suggestions,
+  } = usePlacesAutocomplete({
+    requestOptions: { componentRestrictions: { country: PLACES_COUNTRY_RESTRICTIONS } },
+    debounce: PLACES_DEBOUNCE_DELAY,
+  });
+
+  const [autocompleteValue, setAutocompleteValue] =
+    useState<google.maps.places.AutocompletePrediction | null>(null);
+  const [autocompleteError, setAutocompleteError] = useState<string | null>(null);
+
+  const handleAutocompleteChange = async (
+    _event: SyntheticEvent<Element, Event>,
+    newValue: google.maps.places.AutocompletePrediction | null
+  ): Promise<void> => {
+    setAutocompleteValue(newValue);
+
+    if (!newValue) {
+      return;
+    }
+
+    const { place_id: placeId, description } = newValue;
+
+    try {
+      await getPlaceIdDetails(placeId)
+        .then((data) => {
+          if (!Object.values(data).every((value) => Boolean(value))) {
+            setAutocompleteError(translate('createAppointmentFourth.notPreciseLocation'));
+            return;
+          }
+
+          setLocation(description, false);
+          onLocationChange(data);
+          setAutocompleteError(null);
+        })
+        .catch(() => {
+          setAutocompleteError(translate('createAppointmentFourth.addressDetailsFailed'));
+        });
+    } catch (error) {
+      throw new Error(error);
+    }
+  };
+
+  const handleInputChange = (
+    _event: SyntheticEvent<Element, Event>,
+    newInputValue: string
+  ): void => {
+    setLocation(newInputValue);
+  };
+
+  return (
+    <>
+      <Autocomplete
+        id="google-map-places"
+        getOptionLabel={(option): string =>
+          typeof option === 'string' ? option : option.description
```

Comment:
Better not compare with string

Reply:
typeof operator in TS is completely type safe